Automattic/wp-calypso
# Magic Login: Control flow state with redux instead of separate routes

This PR moves to redux actions to control state & converges on the `/login` route (instead of `/login/send-me-a-link`, for example).

## Manual Testing

* Apply server-side changes (see: D5031-code)
* Visit the [login page](http://calypso.localhost:3000/login)
* Click the `Email me a login link` link
* Follow the flow

## Automated Testing
`npm run test-client -- --grep "state login"`

For item `Initiate state changes via Redux actions` in #12677

## TODOs (in follow up PR(s))

* "reduxify" auth handler components (+ tests)
* component / rendering tests
